pandas/tests/test_algos.py

```diff
@@ -287,6 +287,23 @@ def test_complex_sorting(self):
 
         self.assertRaises(TypeError, algos.factorize, x17[::-1], sort=True)
 
+    def test_uint64_factorize(self):
+        data = np.array([2**63, 1, 2**63], dtype=np.uint64)
```

Comment:
Will do once we can figure what to do about the situation with passing in `NaN` and specifying an integer index.